core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -255,6 +256,47 @@ class TransactionCoordinator(brokerId: Int,
     }
   }
 
+  def handleDescribeTransactions(
+    transactionalId: String
+  ): DescribeTransactionsResponseData.TransactionState = {
+    val transactionState = new DescribeTransactionsResponseData.TransactionState()
+      .setTransactionalId(transactionalId)
+
+    if (!isActive.get()) {
+      transactionState.setErrorCode(Errors.COORDINATOR_NOT_AVAILABLE.code)
+    } else if (transactionalId == null || transactionalId.isEmpty) {
```

Comment:
I agree it seems inconsistent. The protocol definition does not allow it to be nullable, but I think I added the check because the method is one step removed from the request. Perhaps I can change it to raise `IllegalArgumentException` if the `transactionalId` is null.